source/Subsystem/JobSystem.cpp

```diff
@@ -1,103 +1,123 @@
 #include "JobSystem.h"
 
+#include <algorithm>
 #include <chrono>
+#include <functional>
 
-#include "../Utility/Timer.h"
-#include "../Utility/Logger.h"
 #include "../Utility/Assert.h"
+#include "../Utility/Logger.h"
+#include "../Utility/Timer.h"
 
-auto exampleJob = []() -> bool
-{
-	std::this_thread::sleep_for(std::chrono::milliseconds(100));
-
-	return true;
-};
-
-JobSystem::JobSystem(){}
-
-JobSystem::~JobSystem(){}
-
-bool JobSystem::Init()
+JobSystem::JobSystem()
+	: myNeedsUpdate(true)
 {
 	// Leave one thread open for the OS (common decency).
-	myThreadCount = std::thread::hardware_concurrency() - 1;
-	myJobs.reserve(myThreadCount);
-	for (unsigned int i = 0; i < myThreadCount; ++i)
-	{
-		myJobs.emplace_back();
-	}
-
+	const int myThreadCount = std::thread::hardware_concurrency() - 1;
```

Comment:
Maybe want to save this in the class as a member variable. Maybe this information could be shared with other data objects?

Reply:
I'd like to keep the thread system abstracted away from anyone except who works in the job system, so I'll keep it as is for now, might pull it out to the header later.